GeometricAlgorithmsGame/Assets/Scripts/Camera.cs

```diff
@@ -52,10 +56,135 @@ void Update()
     /// </summary>
     /// <param name="floorplan"></param>
     /// <returns></returns>
-    public async Task<IEnumerable<PolygonVertex>> CalculateView(Floorplan floorplan)
+    public IEnumerable<PolygonVertex> CalculateView(Floorplan floorplan)
     {
-        // TODO: To be implemented by Damian M. Buzink
-        throw new NotImplementedException();
+        //Method is sometimes called while the postion is not yet set, just return null for now
+        if(Position == null)
+        {
+            return null;
+        }
+
+        //These will cause issues if the the angle of view causes the min/max angle to flip arround 0/360 degrees.
+        //Rotate it 90 degrees since this script assumes 0 degrees to to the dead right instead of up (Might want to change this to make it consistent)
+        double minAngle = ((Angle + 90) % 360 - _angleOfView / 2) * Mathf.Deg2Rad;
+        double maxAngle = ((Angle + 90) % 360 + _angleOfView / 2) * Mathf.Deg2Rad;
+
+        //We first need a list of all edges
+        var edges = floorplan.SimplePolygon
+            .GetVerticesPairWise()
+            .Select(x => new Edge(new PolygonVertex(x.v1.X, x.v1.Y, new List<PolygonVertex>()), new PolygonVertex(x.v2.X, x.v2.Y, new List<PolygonVertex>())));
+
+        //Get all vertices, group them on their angle to the camera and sort them on their angle
+        var GroupedVertices = edges
+                .SelectMany(x => new List<Tuple<Vertex, Edge>>() { new Tuple<Vertex, Edge>(x.StartPoint, x), new Tuple<Vertex, Edge>(x.EndPoint, x) })
```

Comment:
`new Tuple<Vertex, Edge>(x.EndPoint, x)` is wat overbodig aangezien `(v.EndPoint, x)`. Van mij hoef je het niet aan te passen, maar dat je het weet voor in het vervolg ;)